sotodlib/mapmaking/demod_mapmaker.py

```diff
@@ -452,32 +452,29 @@ def setup_demod_map(noise_model, shape=None, wcs=None, nside=None,
                                          singlestream=singlestream)
     return mapmaker
 
-def write_demod_maps(prefix, data, info, unit='K', split_labels=None, atomic_db=None):
+def write_demod_maps(prefix, data, info, unit='K', split_labels=None):
     """
     Write maps from data into files
     """
     Nsplits = len(split_labels)
     for n_split in range(Nsplits):
-        if np.all(data.wmap[n_split] == 0.0):
-            if atomic_db is not None:
-                smutils.atomic_db_aux(atomic_db, info[n_split], valid=False)
+        info[n_split]["valid"] = np.any(data.wmap[n_split] != 0.0)
+        if info[n_split]["valid"] == False:
             continue
```

Comment:
Sorry, I mis-understood the control flow here. 
```suggestion
        if not np.any(data.wmap[n_split] != 0.0):
            continue
        
        info[n_split]["valid"] = True
```